File: decompiled/sources/com/amo/demo/wheelview/NumericWheelAdapter.java
package com.amo.demo.wheelview;

public class NumericWheelAdapter implements WheelAdapter {
    public static final int DEFAULT_MAX_VALUE = 9;
    private static final int DEFAULT_MIN_VALUE = 0;
    private String format;
    private int maxValue;
    private int minValue;

    public NumericWheelAdapter() {
        this(0, 9);
    }

    public NumericWheelAdapter(int i, int i2) {
        this(i, i2, (String) null);
    }

    public NumericWheelAdapter(int i, int i2, String str) {
        this.minValue = i;
        this.maxValue = i2;
        this.format = str;
    }

    public String getItem(int i) {
        if (i < 0 || i >= getItemsCount()) {
            return null;
        }
        int i2 = this.minValue + i;
        String str = this.format;
        if (str == null) {
            return Integer.toString(i2);
        }
        return String.format(str, new Object[]{Integer.valueOf(i2)});
    }

    public int getItemsCount() {
        return (this.maxValue - this.minValue) + 1;
    }

    public int getMaximumLength() {
        int length = Integer.toString(Math.max(Math.abs(this.maxValue), Math.abs(this.minValue))).length();
        return this.minValue < 0 ? length + 1 : length;
    }
}

File: decompiled/sources/com/amo/demo/wheelview/NumericWheelAdapter.kt
package com.amo.demo.wheelview

import java.util.Locale

class NumericWheelAdapter : WheelAdapter {
    private var format: String?
    private val maxValue: Int
    private val minValue: Int

    constructor() : this(DEFAULT_MIN_VALUE, DEFAULT_MAX_VALUE)

    constructor(minValue: Int, maxValue: Int) : this(minValue, maxValue, null)

    constructor(minValue: Int, maxValue: Int, format: String?) {
        this.minValue = minValue
        this.maxValue = maxValue
        this.format = format
    }

    override fun getItem(i: Int): String? {
        if (i < 0 || i >= getItemsCount()) {
            return null
        }
        val value = minValue + i
        val valueFormat = format
        if (valueFormat == null) {
            return Integer.toString(value)
        }
        return String.format(Locale.getDefault(), valueFormat, value)
    }

    override fun getItemsCount(): Int {
        return (maxValue - minValue) + 1
    }

    override fun getMaximumLength(): Int {
        val length = Integer.toString(maxOf(kotlin.math.abs(maxValue), kotlin.math.abs(minValue))).length
        return if (minValue < 0) {
            length + 1
        } else {
            length
        }
    }

    companion object {
        const val DEFAULT_MAX_VALUE: Int = 9
        private const val DEFAULT_MIN_VALUE: Int = 0
    }
}
